src/plugins/kibana/public/settings/sections/indices/add_data_steps/pipeline_setup/lib/pipeline.js

```diff
@@ -108,24 +107,44 @@ export default class Pipeline {
     return result;
   }
 
-  // Updates the state of the pipeline and processors with the results
-  // from an ingest simulate call.
-  applySimulateResults(results) {
-    //update the outputObject of each processor
-    results.forEach((result) => {
+  _updateProcessorOutputs(simulateResults) {
+    simulateResults.forEach((result) => {
       const processor = this.getProcessorById(result.processorId);
 
       processor.outputObject = _.get(result, 'output');
       processor.error = _.get(result, 'error');
     });
+  }
 
-    //update the inputObject of each processor
-    results.forEach((result) => {
-      const processor = this.getProcessorById(result.processorId);
+  //Updates the error state of the pipeline and its processors
+  //If a pipeline compile error is returned, lock all processors but the error
+  //If a pipeline data error is returned, lock all processors after the error
+  _updateErrorState() {
```

Comment:
Per a recommendation from @epixa, I am going to do the following:
- move the functions so that they are siblings to the class definition in the module. This will keep them from being exposed because they will not be exported
- modify them so that they accept a pipeline argument so that they work in this new context.

This keeps them in the only module that consumes them, and allows me to separate the logic into meaningful, manageable  chunks.

The thing that I don't like about this approach is that it adds additional complexity, where one of my goals was to reduce that. But this is all about tradeoffs, right? :)